src/taco-remote-lib/common/builder.ts

```diff
@@ -123,13 +124,98 @@ class Builder {
             fs.mkdirSync("platforms");
         }
 
+        return this.ensurePlatformAdded().then((addedPlatform) => {
+            if (addedPlatform) {
+                // Note that "cordova platform add" eventually calls "cordova prepare" internally, which is why we don't invoke prepare ourselves when we add the platform.
+                return Q({});
+            }
+
+            return this.update_platform();
+        });
+    }
+
+    /**
+     * Adds the platform if it isn't already present, or removes and re-adds it if the project Cordova version changed, or config.xml has a saved platform version that differs from the installed one.
+     * Returns true if the platform was added, false if not.
+     */
+    private ensurePlatformAdded(): Q.Promise<boolean> {
         if (!fs.existsSync(path.join("platforms", this.currentBuild.buildPlatform))) {
             Logger.log("cordova platform add " + this.currentBuild.buildPlatform);
+
             // Note that "cordova platform add" eventually calls "cordova prepare" internally, which is why we don't invoke prepare ourselves when we add the platform.
-            return this.cordova.raw.platform("add", this.currentBuild.buildPlatform);
-        } else {
-            return this.update_platform();
+            return this.cordova.raw.platform("add", this.currentBuild.buildPlatform).then(() => {
+                return Q(true);
+            });
+        }
+
+        var xmlVersion: string = this.getConfigXmlPlatformVersion();
+
+        return this.getCurrentPlatformVersion().then((installedVersion: string) => {
+            if (xmlVersion) {
+                if (semver.neq(xmlVersion, installedVersion)) {
```

Comment:
I think this is wrong. I think you want to use `semver.satisfies` to let it match ranges, and it looks like it expects `semver.satisfies(singleVersion, versionSpecifier)`. As it stands, if `xmlVersion` is something like `~1.1.0` it'll complain.